File: programming-languages/c/cs50/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

int main(void)
{
    string text = get_string("Write an text: ");
    int sentences = 0;
    int words = 0;
    int chars = strlen(text);
    float average_chars_per_word = 0;
    float average_word_per_sentence = 0;

    for (int i = 0; i < chars; i++) {
        if (text[i] == ' ')
        {
            words++;
            average_chars_per_word =  i / words;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            words++;
            average_chars_per_word =  i / words;

            sentences++;
            average_word_per_sentence = words / sentences;
        }
    }

    int index = (0.0588 * average_chars_per_word * 100) - (0.296 * sentences/words * 100) - 15.8;
    printf("Index is %i\n", index);
}

File: programming-languages/python/cs50/readability.py

text = input("Text: ")

sentences = 0
words = 0
chars = len(text)

avgchrw = 0
avgwrds = 0

for i in range(chars):
    if text[i] == " ":
        words += 1
        avgchrw = i / words
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1
        avgwrds = words / sentences

index = (0.0588 * avgchrw * 100) - (0.296 * sentences/words * 100) - 15.8
print(f"Index is: {index}")
